discord/interactions.py

```diff
@@ -650,7 +786,33 @@ def type(self) -> Optional[InteractionResponseType]:
         """:class:`InteractionResponseType`: The type of response that was sent, ``None`` if response is not done."""
         return self._response_type
 
-    async def defer(self, *, ephemeral: bool = False, thinking: bool = False) -> None:
+    @overload
+    async def defer(
+        self,
+        *,
+        ephemeral: bool = ...,
+        thinking: bool = ...,
+        with_response: Literal[True] = True,
+    ) -> InteractionCallback[ClientT]:
+        ...
+
+    @overload
+    async def defer(
+        self,
+        *,
+        ephemeral: bool = ...,
+        thinking: bool = ...,
+        with_response: Literal[False] = ...,
+    ) -> None:
+        ...
```

Comment:
This overload is wrong, as the `= ...` implies that you are not providing an argument (thus relying on the default), which in this case would be `True`, not the implied `False` here.


```suggestion
    @overload
    async def defer(
        self,
        *,
        ephemeral: bool = ...,
        thinking: bool = ...,
        with_response: Literal[False] = False,
    ) -> None:
        ...
```